File: 02.Array_problems/RunningSum.java

// Problem link - https://leetcode.com/problems/running-sum-of-1d-array/description/

package Array_problems;

/**
 * RunningSum
 */
public class RunningSum {
    public int[] runningSum(int[] nums) {
        int ans[] = new int[nums.length];
        int curSum = 0;
        for(int i=0; i<nums.length; i++){
            curSum += nums[i];
            ans[i] = curSum;
        }
        return ans;
    }
}

File: Array_problems/RunningSum.js

// Problem link - https://leetcode.com/problems/running-sum-of-1d-array/description/

// Approach - 1 

let runningSum = function(nums) {
    let curSum = 0;
    let res = [];
    for(let i=0; i<nums.length; i++){
        curSum += nums[i];
        res.push(curSum);
    }
    return res;
};